flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -126,6 +162,15 @@ public ChannelStateWriter getChannelStateWriter() {
 		return channelStateWriter;
 	}
 
+	@VisibleForTesting
+	int getAbortedCheckpointSize() {
+		return abortedCheckpointIds.size();
+	}
+
+	private boolean checkpointAlreadyAborted(long checkpointId) {
+		return abortedCheckpointIds.remove(checkpointId);
```

Comment:
This is `O(n)`. 
I [suggested](https://github.com/apache/flink/pull/8693#discussion_r425881776) to use `LinkedHashMap`.

Reply:
Thanks for your remind.
I'll use a set based on `LinkedHashMap` to implement a set which would remove oldest checkpoint id automatically.
~~~java
private Set<Long> createAbortedCheckpointSetWithLimitSize(int maxRecordAbortedCheckpoints) {
	return Collections.newSetFromMap(new LinkedHashMap<Long, Boolean>() {
		private static final long serialVersionUID = 1L;

		@Override
		protected boolean removeEldestEntry(Map.Entry<Long, Boolean> eldest) {
			return size() > maxRecordAbortedCheckpoints;
		}
	});
}
~~~